hotfix/TINY-6546: Wendel's feedback on text pattern (#1726)
tinymce/tinymce-docs: plugins/textpattern.md

There are three types of patterns: `inline`, `block`, and `replacement` patterns. Inline patterns have a `start` and an `end` text pattern whereas the block and replacement patterns only have a `start`. A user can specify the formats to be applied to the selection, commands to be executed, or text to be replaced.

> > **Note**: When using list commands make sure that the [lists plugin]({{ site.baseurl }}/plugins/lists) is registered for normalized behavior across browsers.

Reviewer: ```suggestion
```
Added to suggestion below. Sorry

> &nbsp;

Reviewer 2: I think this may have been added accidentally?

Author: Nope, its actually to give a new line so the two notes don't get concatenated in one. I'm not sure of a better solution :-(

Author: Done as bellow comment.

Reviewer 2: I realise this has changed anyway, but for future reference I'm pretty sure you just do a newline in markdown to split them, no need for the `&nbsp;`
```
> **Note**: Some text

> **Note**: Some text in a new block
```

Author: It won't work @JamesToohey  :-). Me and Tyler tried that. It concatenates them. It may be the markdown linter that we use that auto-fix things when I save the file. It gets rid of the extra line, so one sits right beside the other which 'concatenate" them :-)

> &nbsp;

Reviewer: ```suggestion
```
Not needed. I wish it was this easy to concat notes

Author: When I actually remove this, the two notes will look like this: 

![Screen Shot 2020-10-22 at 1 29 30 pm](https://user-images.githubusercontent.com/6000860/96821119-a5761400-146a-11eb-9497-bb55e3286e8b.png)

Author: as one comes after the other:

![Screen Shot 2020-10-22 at 1 30 52 pm](https://user-images.githubusercontent.com/6000860/96821216-d22a2b80-146a-11eb-94ed-5711c03f3d7e.png)

Author: Wait... got it working: 

![Screen Shot 2020-10-22 at 1 32 44 pm](https://user-images.githubusercontent.com/6000860/96821392-1d443e80-146b-11eb-8457-f4c7e4071df6.png)

![Screen Shot 2020-10-22 at 1 33 19 pm](https://user-images.githubusercontent.com/6000860/96821420-29300080-146b-11eb-813c-b85ede63e7a0.png)

> > **Note**: Formats and commands must be already registered with the editor. See the [formats]({{ site.baseurl }}/configure/content-formatting/#formats) and [commands]({{ site.baseurl }}/api/tinymce/tinymce.editorcommands/) documentation for more information.

Reviewer: ```suggestion
> **Important**: Any formats or commands used by `textpatterns` need to be registered with the editor when it is initialized. This may include enabling relevant plugins, such as the `lists` plugin. 
> For information on:
>
> - Registering formats for {{site.productname}}, see: [Content formatting options - `formats`]({{site.baseurl}}/configure/content-formatting/#formats).
> - Registering commands for {{site.productname}}, see: [{{site.productname}} APIs - EditorCommands]({{site.baseurl}}/api/tinymce/tinymce.editorcommands/).
```

  {start: '####', format: 'h4'},
  {start: '#####', format: 'h5'},
  {start: '######', format: 'h6'},

>   {start: '1. ', cmd: 'InsertOrderedList'}, // these configurations will need list plugin

Reviewer 2: ```suggestion
  {start: '1. ', cmd: 'InsertOrderedList'}, // these configurations will need the lists plugin
```

>   {start: '1. ', cmd: 'InsertOrderedList'}, // these configurations will need the lists plugin

Author: Done

```js
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML

>   plugin: 'textpattern link', // link is needed here for the following example to work correctly

Reviewer 2: For consistency?
```suggestion
  plugin: 'textpattern link', // the link plugin is needed here for the following example to work correctly
```

Reviewer: ```suggestion
  // The `link` plugin is required for the `createLink` command
  plugin: 'textpattern link',
```

Author: Done

Author: Done Tyler's comment.

```js
tinymce.init({
  selector: 'textarea', // change this value according to your HTML

>   // make sure to have the required plugins here,

Reviewer 2: This is mentioned a few times now, I wonder whether it might be a bit of an overload here?

Reviewer: Maybe
```
  // The `lists` plugin is required for list-related text patterns
```
And delete the rest of the comment?

Author: Any more thoughts on this one, please?

Author: Done

Using the configuration in this example:

* `{start: '#', format: 'h1'}` - Typing `#`, some text, and then pressing `Enter` will convert the text to a `h1` heading.

> * Typing `1.` followed by a space, your desired text, and then pressing `Enter` will convert the text into an ordered list, with the original text as the first list item, and the new line as the second list item. Since we have specified `value`, this pattern will execute `editor.execCommand('InsertOrderedList', false, { 'list-style-type': 'decimal'})`.

Reviewer: ```suggestion
* Typing `1.` followed by a space, the desired text, and then pressing `Enter`; the editor will convert the text into an ordered list, with the original text as the first list item, and the new line as the second list item. Since we have specified `value`, this pattern will execute `editor.execCommand('InsertOrderedList', false, { 'list-style-type': 'decimal'})`.
```

> **Note**: Replacement patterns are executed on either pressing the **spacebar** or the **Enter** key.

> 

Reviewer: test comment 

- [ ] Delete more space
